pkg/api/controller.go

```diff
@@ -1611,6 +1608,11 @@ func (c *Controller) ensureStorageNamespace(ctx context.Context, storageNamespac
 		dummyData = "this is dummy data - created by lakeFS in order to check accessibility"
 	)
 
+	storagePrefixRegex := regexp.MustCompile(c.BlockAdapter.GetStorageNamespaceInfo().ValidityRegex)
```

Comment:
Prefer not to do "must compile" at runtime:
Check if you can use https://pkg.go.dev/regexp#MatchString
Alt. check if compile failed